Samples/src/IntroToIQSharp/Notebook.ipynb

```diff
@@ -56,138 +90,425 @@
+    "- BigInt\n",
+    "- Bool\n",
+    "- Double\n",
+    "\n",
+    "but also some quantum-specific:\n",
+    "- Qubit\n",
+    "- Result\n",
+    "- Pauli\n",
+    "\n",
+    "The complete set of primitive types can be found in [Q# type model documentation](https://docs.microsoft.com/en-us/quantum/language/type-model?view=qsharp-preview#primitive-types).\n",
+    "\n",
+    "In classical programs a bit, or binary digit, represents the basic unit of information and can be only one of two states: `0` or `1`.\n",
+    "\n",
+    "Similarly, the **qubit**, or quantum bit, represents the basic unit of information for quantum algorithms. Like a bit, a qubit when measured can only be in one of two states:  $|0\\rangle$ or  $|1\\rangle$; however during execution, the state of the qubit represents *the probability of reading* $|0\\rangle$ or $|1\\rangle$. \n",
+    "\n",
+    "> For more information about qubits and their mathematical description, \n",
+    "> take a look at [the qubit topic](https://docs.microsoft.com/en-us/quantum/concepts/the-qubit?view=qsharp-preview) inside \n",
     "\n",
-    "If the compiler detects any errors, it will instead show the list of errors in the output. For example:"
+    "In Q#, $|0\\rangle$ and $|1\\rangle$ are represented by `Result.Zero` and `Result.One` accordingly and qubits can only be allocated inside an operation with the `using` statement. When it gets allocated a qubit is always in the $|0\\rangle$  state. To measure a qubit and read its value you use the [`M`](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.primitive.m) intrinsic operation.\n",
+    "\n",
+    "As such, the next `HelloQuantum` operation always return `Result.Zero`:"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 2,
-   "metadata": {},
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
    "outputs": [
     {
-     "name": "stderr",
-     "output_type": "stream",
-     "text": [
-      "/snippet:(5,5): error QS5022: No identifier with that name exists.\n",
-      "/snippet:(9,12): error QS6211: Unexpected argument tuple. Expecting an argument of type String.\n"
-     ]
+     "data": {
+      "application/json": "[\"HelloQuantum\"]",
+      "text/html": [
+       "<ul><li>HelloQuantum</li></ul>"
+      ],
+      "text/plain": [
+       "HelloQuantum"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
     }
    ],
    "source": [
-    "operation InvalidQ() : Unit {\n",
-    "\n",
-    "    // The `FooBar` operation doesn't exist, so the following line\n",
-    "    // will generate a `No variable with that name exists.` error:\n",
-    "    FooBar(\"Hello again!\");\n",
-    "    \n",
-    "    // `Message` takes only one string argument, so the following line\n",
-    "    // will generate a `Unexpected argument tuple.` error:\n",
-    "    Message(1, 2);\n",
+    "operation HelloQuantum() : Result {\n",
```

Comment:
In keeping with the style guide, I'd suggest naming this operation as a verb. That will help users learn the naming conventions that we use throughout the libraries, for instance.